http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/AppRoutes.scala

```diff
@@ -19,154 +19,156 @@ import org.broadinstitute.dsde.workbench.leonardo.http.api.AppRoutes._
 import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
 import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.ServiceName
 import org.broadinstitute.dsde.workbench.leonardo.service.KubernetesService
+import io.opencensus.scala.akka.http.TracingDirective.traceRequestForService
 
-class AppRoutes(kubernetesService: KubernetesService[IO], userInfoDirectives: UserInfoDirectives)(
-  implicit timer: Timer[IO]
-) {
-
-  val routes: server.Route = userInfoDirectives.requireUserInfo { userInfo =>
-    CookieSupport.setTokenCookie(userInfo, CookieSupport.tokenCookieName) {
-      pathPrefix("google" / "v1" / "apps") {
-        pathEndOrSingleSlash {
-          parameterMap { params =>
-            get {
-              complete(
-                listAppHandler(userInfo, None, params)
-              )
-            }
-          }
-        } ~
-          pathPrefix(googleProjectSegment) { googleProject =>
+class AppRoutes(kubernetesService: KubernetesService[IO], userInfoDirectives: UserInfoDirectives) {
+
+  val routes: server.Route = traceRequestForService(serviceData) { span =>
```

Comment:
I was wondering whether `traceRequestForService` could be at the top-level, or if it should be done per-route. I guess it should be okay here.